src/util-decode-mime.c

```diff
@@ -2655,24 +2661,18 @@ static int TestDataChunkCallback(const uint8_t *chunk, uint32_t len,
 
     /* Add up the line counts */
     if (len > 0) {
-
-        uint8_t *remainPtr;
-        uint8_t *tok;
-        uint32_t tokLen;
-
+        if ((*line_count) == 0) {
+            (*line_count)++;
+        }
         PrintChars(SC_LOG_DEBUG, "CHUNK", chunk, len);
-
-        /* Parse each line one by one */
-        remainPtr = (uint8_t *) chunk;
-        do {
-            tok = GetLine(remainPtr, len - (remainPtr - (uint8_t *) chunk),
-                    &remainPtr, &tokLen);
-            if (tok != NULL && tok != remainPtr) {
+        for (uint32_t i = 0; i < len; i++) {
+            if (chunk[i] == CR || chunk[i] == LF) {
+                if (i + 1 < len && (chunk[i + 1] == CR || chunk[i + 1] == LF)) {
+                    i++;
```

Comment:
this increases `i` for CRLF LFCR LFLF CRCR, right? Is that intentional?

Reply:
Indeed `GetLine` adds another `chunk[i] != chunk[i + 1]` condition